### PR TITLE
We can now set image descriptions via API

### PR DESCRIPTION
**Pull request by annando**
When posting pictures to Friendica by using the API, we now can add alternative descriptions to it.

### CODE REVIEW

**Comment by nupplaphil** [comment]
Please add a test after this
https://github.com/friendica/friendica/blob/4b60e89c0719c2ac7bfa39e900c356efdc7a0f32/tests/include/ApiTest.php#L1333-L1354
You can take this test case as a blueprint for a test like `testApiMediaMetadataCreate()`

**Comment by annando** [comment]
@nupplaphil How can I send data in that test script so that it appears in ```php://input```?

**Comment by nupplaphil** [comment]
As far as I can see, the other methods for uploading files are using the global `$_FILES` (https://www.php.net/manual/en/reserved.variables.files.php) or `$_POST` , wouldn't it be better to use it for this method too? I'm not sure if this creates the same result - pinging our PHP-code-guru @mrpetovan 

Because with `$_FILES`, you can than override it for each test case

**Comment by nupplaphil** [comment]
@annando I found an example for `json_decode` in another API function:
https://github.com/friendica/friendica/blob/79bc773d020175255b2957b54849d686e48373da/include/api.php#L5791-L5809
Maybe this help :-)

**Comment by annando** [comment]
To stay compatible with the existing clients like Twidere we have to stick to ```php://input```. And values that are transferred this way don't appear in any variable like ```$_REQUEST```, ```$_POST```, ```$_GET``` or ```$_FILES```.

**Comment by annando** [comment]
I guess the only possible way to work around this problem is to write an own function that normally does return the value of ```php://input``` and in test mode (and only there!) would return the content of some special environment variable.

But this needs to be done and tested. So I would suggest merging this PR. After this we could come with a solution for this problem.

File: doc/api.md [hunk 1]
```diff
@@ -393,6 +393,27 @@ Object of:
...
+
+#### Parameters
+
+Parameters are sent as JSON object:
+
+```
+{
+	"media_id":"1234",
+	"alt_text": {
+		"text":"Here comes the description"
```
Comment (MrPetovan) [inline doc/api.md]: Is it a standard data structure?

Comment (annando) [reply doc/api.md]: What do you mean with this? In fact I did the coding in reference to this documentation: https://developer.twitter.com/en/docs/media/upload-media/api-reference/post-media-metadata-create.html

**Comment by annando** [comment]
I just encapsulated the postdata stuff into a single function. I guess with some class magic this can now behave differently while testing.

File: doc/api.md (hunk 1 above)
Comment (MrPetovan) [reply doc/api.md]: That was my question, thanks.

**Pull request merged**